### ISSUE TITLE
Show all Subscale Data on Visualization 

### ISSUE THREAD

**Issue by WorldImpex** [ChildMindInstitute/mindlogger-admin-OLD-Vue#938]
As a manager or reviewer, I want to see previous subscale results so that I can compare the newest data set with older data sets 

Requirements
- Ensure that the compliance chart shows a dot for every time a user submits an activity 
![Screen Shot 2021-06-03 at 6.50.53 AM.png](https://images.zenhubusercontent.com/5e1336f1b24025a7b2654475/93f9c604-7a49-494d-9ba9-de93dec656f5)
- Allow the user to click on the dot and show a drop-down with "Show Subscale Results" 
- If the user selects "Show Subscale Results" then update the subscale tables to show the results from that data entry point. 
- Show a selection around the compliance dot so the user knows the information is being shown on the visualization is from that point 
- Export all subscale data collected

**Comment by natalia-muzyka** [comment]
Blocked by 
[The "Average of Item Scores" isn't displayed selected after saving and reentering the edit subscale page #382](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-applet-builder/382)

**Comment by natalia-muzyka** [comment]
1) [#1065 The compliance chart dots within the same date are overlapping](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1065)
https://www.screencast.com/t/JcefO828NzJc

2) [#1801 The selected in the assessment responses don't match the DataViz displaying](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-app/1801)
Video: https://www.screencast.com/t/8v8kKmeUYYfu
Compare table mobile app & dataviz: https://www.screencast.com/t/gzJz7t9SyA

3) [#1066 After selecting the "Show Subscale Results" all the results are shown instead of the results from the selected data entry point](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1066)
https://www.screencast.com/t/A9sJdKG9h

@WorldImpex  how the export of all subscale data should be presented?

**Environment:**
https://admin-staging.mindlogger.org/
Account: geria.test@yahoo.com // 3FuuFz60
Applet: Retest subscale 2 (Applet password: Qwe123!!!)

**Comment by WorldImpex** [comment]
@natalia-muzyka It should be part of the CSV export with all of the details like score achieved and message shown

**Comment by natalia-muzyka** [comment]
Update 24/06

1) The dots(responses) are displayed reverted within a day timeline
The first selected option is displayed as the last one and so on
https://www.screencast.com/t/FhNN6s7reK

2) Created a new ticket:
[[UI] DataViz issues#1089](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1089)

3) Created new issue: 
[The time of taking an assessment isn't displayed correctly in all of the points:](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-app/1807)
the "review" tab matches the real-time of taking an assessment **but** doesn't match the "responses" tab and exported CSV file
https://www.screencast.com/t/8TPqaInFp

4) Created new issue: 
[After selecting the Total score calculation (Average or Sum) the DataViz can't be loaded. ](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1078)Noticed that after activating any of the checkboxes there is the lookup table automatically added on Total score calculation (the lookup table btn is active after saving and refreshing the page), but after opening there is no default table.
https://www.screencast.com/t/VvhsIvkVJBs
https://www.screencast.com/t/pebAMvb0nGX

**Environment:**
https://admin-staging.mindlogger.org/
Account: geria.test@yahoo.com // 3FuuFz60
Applet: Retest subscale 2 (Applet password: Qwe123!!!)

Still need to be fixed:
#1066 After selecting the "Show Subscale Results" all the results are shown instead of the results from the selected data entry point
https://www.screencast.com/t/A9sJdKG9h

**Comment by natalia-muzyka** [comment]
@WorldImpex Is this an acceptable score displaying or it should be displayed shortened?
https://www.screencast.com/t/f9PBDbsM

**Comment by WorldImpex** [comment]
@natalia-muzyka I think 3 decimals should be sufficient for this

**Comment by natalia-muzyka** [comment]
@mtg137 On macOS 10.15 // macOS 11 // Safari after clicking the compliance chart dot:
- the dropdown isn't shown up 
- the selection isn't shown up
Not reproducible on Chrome.

Video: https://www.screencast.com/t/QtHg6vZwPdu

**Comment by natalia-muzyka** [comment]
Fixed:
- dropdown shows up 
- selection shows up

New issues for Safari:
1) The "Show subscales result" option is hidden (out of workspace, can't be scrolled down) for the latest activity after clicking on the compliance chart dot of the collapsed activity:
https://www.screencast.com/t/xkrzWB4mwf

2) The additional info isn't shown up after clicking the subscale on the chart
https://www.screencast.com/t/hNAJjuBWgn7O

But these can be resolved in the next release scope, I think,
wdyt @WorldImpex @mtg137 ?

**Comment by WorldImpex** [comment]
Yes we should @natalia-muzyka & @mtg137

**Comment by techdev137** [comment]
@natalia-muzyka I was able to reproduce 1).
but I wasn't able to repro 2).

I tested on this account.

geria2.test@gmail.com // McgzF127
applet: subscale (2) (1)
user: aleksey@outlook.com
activity: new lookup table
applet password: Qwe123!!!

could you please let me know account you reproduced 2)?

**Comment by natalia-muzyka** [comment]
@mtg137 the same account, applet, and activity 
dataviz for manager: geria2.test@gmail.com
and it's reproducible only for Safari

here you can see the same subscale chart on two browsers (chrome & safari)
video: https://www.screencast.com/t/XUAV4QM5A

**Comment by techdev137** [comment]
Hmm, this is safari browser as well, but it's working well.

![image.png](https://images.zenhubusercontent.com/5fbce3e9259bafeab239c8c3/7f6d2d52-22b1-4dc1-8ab0-02bc8080fbcb)

@natalia-muzyka perhaps, is there difference because you tested it on anydesk and I tested directly on my device?

**Comment by techdev137** [comment]
@natalia-muzyka what's version of your safari browser?

**Comment by natalia-muzyka** [comment]
@mtg137 
macOS 10.15.6 // Safari 13.1.2 (15609.3.5.1.3)

**Comment by techdev137** [comment]
thanks.

**Comment by techdev137** [comment]
@natalia-muzyka FYI, safari version on my device was lower than yours. that was reason I couldn't reproduce issue.  I'll fix it.

**Comment by natalia-muzyka** [comment]
@mtg137 @WorldImpex  both issues are still reproducible on staging and prod, were the changes deployed to the admin panel?
video: https://www.screencast.com/t/LsSkUnEug
environment:
macOS 10.15 // Safari 13.1.2
macOS 11.0 // Safari 14.0.1

**Comment by WorldImpex** [comment]
@devbtech Was this deployed?

**Comment by natalia-muzyka** [comment]
Safari issues are fixed
video: https://www.screencast.com/t/Qx62Iv0Hbms
environment:
https://admin.mindlogger.org/
macOS 10.15 // Safari 13.1.2
macOS 11.0 // Safari 14.0.1
